tpm2tools/keys_test.go

```diff
@@ -117,6 +118,37 @@ func TestKeyCreation(t *testing.T) {
 	}
 }
 
+func TestCreatePublicAreaFromPublicKey(t *testing.T) {
```

Comment:
Would comparing the name be sufficient for checking for exact public area match, since it's a hash of the public area? For generating a key from the TPM I used EndorsmentKeyRSA() which returns a Key, but Key types don't give public access to their tpm2.Public so I can't easily do a comparison.

Reply:
Name comparison is fine!